### ISSUE TITLE
GeoJSON layers rendering is very slow in firefox on high resolutions

### ISSUE THREAD

**Issue by meelkor** [mapbox/mapbox-gl-js#6661]
On higher resolutions (like 1080p and higher for m machine), it takes quite a long time before vector tiles are updated when zooming or panning. While in webkit browers the tiles appear almost instantly. Once they are cached, they are renderd instantly. It applies for both GeoJSON sources and vector sources.

While waiting for the layers to appear the map is fully responsive. Ironically map zooming is much smoother in firefox because the vector layers are not drawn while zooming I guess.

There is already several performance-related issues about firefox already, but none of those seemed to be relevant to my issue, so hopefully this isn't duplicate.

**mapbox-gl-js version**: 45
**browser**: Mozilla Firefox 60.0

### Steps to Trigger Behavior

 1. Open mapbox-gl on at least 1080p resolution (ideally higher, provided screencap is on 1440p)
 2. Zoom out / pan so some new vector stuff comes in viewport

### Link to Demonstration

Firefox:
https://drive.google.com/open?id=1ezTt2nfr-hpCoME_iQl4ymAuu8Gjw7Ua

Chrome:
https://drive.google.com/open?id=1JSqAaGYWJA-eqQ6o-0XwkXtnDJW_X8FC

http://output.jsbin.com/rurujacewe

### Expected Behavior

GeoJSON layers to be rendered at approximately same speed as raster layer.

### Actual Behavior

It takes several seconds before tiles get updated.

**Comment by mourner** [comment]
Unfortunately I can't reproduce this on Firefox 60.0.2 on an Macbook Pro 15'' Retina when opening the example full-screen. What operating system do you use? Does it reproduce on other machines / OS?

**Comment by meelkor** [comment]
The screencap I've posted was from arch linux, but I believe it behaved the same on Windows 10 as well. I should have also mentioned that the issue isn't that strong on machines with dedicated GPU. Also not sure what's the actual resolution on Macbook Pro, is it actually that high?

I've just tried it on my desktop with windows 10 and GeForce 970 on 1440p and the issue is pretty less notable, but it's still there. But on computers with integrated GPU it makes data visualization quite slow. On chrome the vector layers get rendered right away on lower FPS.

https://drive.google.com/file/d/1FSkD7-raWuX_tLYebZeSM8Vtr6VYV6iO/view

For now we've decided to create custom overlay and visualize the data on that one as a workaround.
